src/language-js/print/ternary.js

```diff
@@ -158,6 +160,79 @@ function printTernaryTest(path, options, print) {
   return printed;
 }
 
+const ancestorNameMap = new Map([
+  ["AssignmentExpression", "right"],
+  ["VariableDeclarator", "init"],
+  ["ReturnStatement", "argument"],
+  ["ThrowStatement", "argument"],
+  ["UnaryExpression", "argument"],
+  ["YieldExpression", "argument"],
+]);
+function shouldExtraIndentForConditionalExpression(path) {
+  const node = path.getValue();
+  if (node.type !== "ConditionalExpression") {
+    return false;
+  }
+
+  const parentName = path.callParent((path) => path.getName());
+  if (!parentName) {
+    return false;
+  }
+
+  let ancestorCount = 1;
+  let ancestor = path.getParentNode(ancestorCount);
+  while (isChainElement(ancestor)) {
+    ancestor = path.getParentNode(++ancestorCount);
+  }
+  const checkAncestor = (node) => {
+    const name = ancestorNameMap.get(ancestor.type);
+    return ancestor[name] === node;
+  };
+
+  const parent = path.getParentNode();
+
+  /**
+   * foo = (
+   *   condition
+   *     ? first
+   *     : second
+   * ) as SomeType;
+   */
+  if (parent.type === "TSAsExpression" && checkAncestor(parent)) {
+    return true;
+  }
+
+  const name = path.getName();
+
+  /**
+   * foo = new (
+   *   condition
+   *     ? first
+   *     : second
+   * )(arguments);
+   */
+  if (
+    parent.type === "NewExpression" &&
+    checkAncestor(parent) &&
+    name === "callee"
+  ) {
+    return true;
+  }
+
+  const chainRoot = path.getParentNode(ancestorCount - 1);
+  if (
+    isChainElement(parent) &&
```

Comment:
`isChainElement` is not needed anymore, following codition already ensure it.

Reply:
And please tweak the conditions order a little bit. check name first and check parent first.